src/libmongoc/src/mongoc/mongoc-client-side-encryption.c

```diff
@@ -117,6 +119,20 @@ mongoc_auto_encryption_opts_set_kms_providers (
    }
 }
 
+void
+mongoc_auto_encryption_opts_set_tls_opts (mongoc_auto_encryption_opts_t *opts,
+                                          const bson_t *tls_opts)
+{
+   if (!opts) {
+      return;
+   }
+   bson_destroy (opts->tls_opts);
+   opts->tls_opts = NULL;
```

Comment:
This is ok, but since it's getting repeated I *wonder* if:

opts_tls_opts = bson_copy_or_null(tls_opts); // makes a copy, or returns NULL
or
bson_copy_or_null(tls_opts, opts->tls_opts); // destroys opts->tls_opts if non-null, then as per above

...might not be a nice simplification-- perhaps with a more catchy name. May or may not be worth doing, but is worth thinking about.

Reply:
I am not confident `bson_copy_or_null` is generically useful enough to add to public API, but it seems reasonable to add as a private helper. Added `_bson_copy_or_null` in this file.

Reply:
I am not confident `bson_copy_or_null` is generically useful enough to add to public API, but it seems reasonable to add as a private helper. Added `_bson_copy_or_null` in this file.